src/components/TradeWidget/TokenRow.tsx

```diff
@@ -137,13 +139,19 @@ function displayBalance(balance: TokenBalanceDetails | undefined | null, key: st
   return formatAmount(balance[key], balance.decimals) || '0'
 }
 
-function getMax(balance: TokenBalanceDetails | null, token: TokenDetails): string {
-  return formatAmountFull((balance || {}).exchangeBalance, token.decimals, false) || '0'
+function adjustPrecision(value: string | undefined | null, precision: number): string {
```

Comment:
What does it do exactly?
shouldn't this be using the utils we have. We have kind of similar code we could use, right?

Reply:
This function `rounds` the value to the required precision, such as `1.15` on precision `1` becomes `1.1`.

The existing functionality (`parseAmount`) returns `null` when decimal places > precision: https://github.com/gnosis/dex-react/blob/develop/src/utils/format.ts#L89

But this is a good point. I can just adapt `parseAmount` instead.

Reply:
Aha, I understand 👍 

Yes, I agree, a refactor there would be a good idea

Reply:
While working on it I realized that it won't be very efficient if I use existing helpers, even after adjusting `parseAmount` to truncate on given precision.
That's because I'll have to do 2 conversions: input string -> BN -> string, something like:

```typescript
function adjustPrecision2(value: string | undefined | null, precision: number): string {
  const bn = parseAmount(value, precision) || ZERO
  return formatAmountFull(bn, precision, false) || value
}
```

I believe the version implemented might be more efficient. I would instead move it to utils and extract common functionality there.